draft-ietf-ppm-dap.md

```diff
@@ -1267,106 +1334,171 @@ This message consists of:
+`inbound` denote the payload of the prep step sent by the Leader):
+
+~~~
+state = VDAF.ping_pong_start(vdaf_verify_key,
+                             False,
+                             agg_param,
+                             report_id,
+                             public_share,
+                             plaintext_input_share.payload)
+if state != Rejected():
+  (state, outbound) = VDAF.ping_pong_resp(agg_param, state, inbound)
+~~~
+
+where:
+
+* `vdaf_verify_key` is the VDAF verification key for the task
+* `agg_param` is the VDAF aggregation parameter sent in the
+  `AggregationJobInitReq`
+* `report_id` is the report ID
+* `public_share` is the report's public share
+* `plaintext_input_share` is the Helper's `PlaintextInputShare`
 
+This procedure determines the initial per-report `state`, as well as the
+initial `outbound` to send in response to the Leader. If `state == Rejected()`,
+then the Helper responds with
+
+~~~
 struct {
-  ReportID report_id;
-  PrepareStepState prepare_step_state;
-  select (PrepareStep.prepare_step_state) {
-    case continued: opaque prep_msg<0..2^32-1>; /* VDAF preparation message */
-    case finished:  Empty;
-    case failed:    ReportShareError;
-  };
+  PrepareStepState prepare_step_state = 3; /* reject */
+  ReportId report_id;
+  ReportShareError report_share_error = 5; /* vdaf_prep_error */
+} PrepareStep;
+~~~
+
+Otherwise the Helper responds with
+
+~~~
+struct {
+  PrepareStepState prepare_step_state = 1; /* continue */
+  ReportId report_id;
+  opaque payload<0..2^32-1> = outbound;
 } PrepareStep;
+~~~
+
+Finally, the Helper creates an `AggregationJobResp` to send to the Leader. This
+message is structured as follows:
 
+~~~
 struct {
   PrepareStep prepare_steps<1..2^32-1>;
 } AggregationJobResp;
 ~~~
 
-The message is a sequence of PrepareStep values, the order of which matches that
-of the ReportShare values in `AggregationJobInitReq.report_shares`. Each report
-that was marked as invalid is assigned the `PrepareStepState` `failed`.
-Otherwise, the `PrepareStep` is either marked as continued with the output
-`prep_msg`, or is marked as finished if the VDAF preparation process is finished
-for the report share. The Helper's aggregation job is now in round 0.
+where `prepare_steps` are the outbound prep steps computed in the previous
+step. The order MUST match `AggregationJobInitReq.prepare_steps`.
 
-On success, the Helper responds to the Leader with HTTP status code 201 Created
-and a body consisting of the `AggregationJobResp`, with media type
+The Helper responds to the Leader with HTTP status code 201 Created and
+a body consisting of the `AggregationJobResp`, with media type
 "application/dap-aggregation-job-resp".
 
+Finally, the Helper stores `state` to prepare for the next aggregation step
+({{aggregation-helper-continuation}}, if the state is `Continued`) or for the
+collection flow ({{collect-flow}}, if the state is `Finished`).
```

Comment:
* Let's parse out the `out_share`, since {{collect-flow}} knows nothing about `state`.
* Missing closed paran
* collect flow -> sub-protocol
* let's try to be as precise as we can, i.e., instead of "aggregation step", which is a bit ill-defined, let's say "continuation step", which we use elsewhere.
```suggestion
Finally, if `state == Continued(prep_state)` then the Helper stores `state` for the next continuation step
({{aggregation-helper-continuation}}). Otherwise, if `state == Finished(out_share)`, then the Helper stores `out_share` for use in the collection sub-protocol ({{collect-flow}}).
```